### PR TITLE
Bug fix for igm deletion when the network import failed

### PR DESCRIPTION
**Pull request by SlimaneAmar**
Signed-off-by: Slimane AMAR <slimane.amar@rte-france.com>

### CODE REVIEW

File: src/test/java/org/gridsuite/merge/orchestrator/server/MergeOrchestratorIT.java [hunk 1]
```diff
@@ -538,6 +551,76 @@ public void testDeleteMerge() {
         assertNull(output.receive(1000));
     }
 
+    @Test
+    public void testDeleteIgmMergeWithImportError() {
+        mergeOrchestratorConfigService.addConfig(new ProcessConfig("FRES_2D", "2D", List.of("FR", "ES"), false));
+
+        // send tsos FR, ES and PT with business process = 2D
```
Comment (FranckLecuyer) [inline src/test/java/org/gridsuite/merge/orchestrator/server/MergeOrchestratorIT.java:558]: Change comment

File: src/test/java/org/gridsuite/merge/orchestrator/server/MergeOrchestratorIT.java [hunk 2]
```diff
@@ -538,6 +551,76 @@ public void testDeleteMerge() {
...
+        assertEquals(0, igmRepository.findAll().size());
+
+        assertNull(output.receive(1000));
+    }
+
+    @Test
+    public void  testDeleteIgmMergeWithValidationFailed() {
+        mergeOrchestratorConfigService.addConfig(new ProcessConfig("FRES_2D", "2D", List.of("FR", "ES"), false));
+
+        // send tsos FR, ES and PT with business process = 2D
```
Comment (FranckLecuyer) [inline src/test/java/org/gridsuite/merge/orchestrator/server/MergeOrchestratorIT.java:593]: Change comment

File: src/main/java/org/gridsuite/merge/orchestrator/server/MergeOrchestratorService.java [hunk 3]
```diff
@@ -144,28 +144,34 @@ public void consume(Message<String> message) {
...
-                    UUID processConfigNetworkUuid = caseFetcherService.importCase(caseUuid, lastBoundaries);
-                    merge(processConfig, dateTime, date, tso, valid, processConfigNetworkUuid, caseUuid, null, null, lastBoundariesUuid);
+                    for (ProcessConfig processConfig : matchingProcessConfigList.subList(1, matchingProcessConfigList.size())) {
+                        // import IGM into the network store
+                        UUID processConfigNetworkUuid = caseFetcherService.importCase(caseUuid, lastBoundaries);
+                        merge(processConfig, dateTime, date, tso, valid, processConfigNetworkUuid, caseUuid, null, null, lastBoundariesUuid);
+                    }
+                } catch (Exception e) {
+                    ProcessConfig processConfig = matchingProcessConfigList.get(0);
+                    mergeEventService.addMergeIgmEvent(processConfig.getProcess(), processConfig.getBusinessProcess(), dateTime, tso, IgmStatus.VALIDATION_FAILED, null, null, null, null, null);
```
Comment (FranckLecuyer) [inline src/main/java/org/gridsuite/merge/orchestrator/server/MergeOrchestratorService.java:173]: Maybe we could distinguish between a failed import in the network store, and a validation that failed on the imported network in the network store, with two different igm status ?

Maybe in another US ....

**Review by FranckLecuyer — CHANGES_REQUESTED** [review]

File: src/test/java/org/gridsuite/merge/orchestrator/server/MergeOrchestratorIT.java (hunk 1 above)
Comment (SlimaneAmar) [reply src/test/java/org/gridsuite/merge/orchestrator/server/MergeOrchestratorIT.java:558]: Done

File: src/test/java/org/gridsuite/merge/orchestrator/server/MergeOrchestratorIT.java (hunk 2 above)
Comment (SlimaneAmar) [reply src/test/java/org/gridsuite/merge/orchestrator/server/MergeOrchestratorIT.java:593]: Done

File: src/main/java/org/gridsuite/merge/orchestrator/server/MergeOrchestratorService.java (hunk 3 above)
Comment (SlimaneAmar) [reply src/main/java/org/gridsuite/merge/orchestrator/server/MergeOrchestratorService.java:173]: Yes we should do (IMPORT_FAILED)
An another US

**Review by FranckLecuyer — APPROVED** [review]

**Review by etiennehomer — APPROVED** [review]

**Review by jonenst — APPROVED** [review]

**Pull request merged**